src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/IVsProjectRestoreInfoSource.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.SolutionRestoreManager
+{
+    /// <summary>
+    /// Represents a package restore service API for integration with a project system.
+    /// Implemented by the project-system.
+    /// </summary>
+    [ComImport]
+    [Guid("35AD5FF2-6AB7-48E9-BCC0-189042410FA6")]
+    public interface IVsProjectRestoreInfoSource
+    {
+        /// <summary>
+        /// Project Unique Name.
+        /// Must be equivalent to the name provided in the <see cref="IVsSolutionRestoreService3.NominateProjectAsync(string, IVsProjectRestoreInfo2, CancellationToken)"/> or equivalent.
+        /// </summary>
+        /// <remarks>Never <see langword="null"/>.</remarks>
+        string Name { get; }
+
+        /// <summary>
+        /// Whether the source needs to do some work that could lead to a nomination.
+        /// This method may be called frequently, so it should be very efficient.
+        /// </summary> 
+        bool HasPendingNomination();
+
+        /// <summary>
+        /// NuGet calls this method to wait on a potential nomination.
+        /// If the project has no pending restore data, it will return a completed task.
+        /// Otherwise, the task will be completed once the project nominates.
+        /// The task will be cancelled, if the source decide it no longer needs to nominate (for example: the restore state has no change)
+        /// The task will be failed, if the source runs into a problem, and it cannot get the correct data to nominate (for example: DT build failed)
+        /// </summary>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        Task WhenNominatedAsync(CancellationToken cancellationToken);
```

Comment:
The XML doc gets rendered similar to HTML, specifically that new lines and whitespace is ignored. Therefore, the summary above will appear in Intellisense as a single line.  Hover the mouse over the method name to see Visual Studio's Intellisense popup, or temporarily in another method try invoking the method, to see what customers/partners calling the API will see. If it's difficult to read, please consider adding formatting to the XMLDoc.

Reply:
There's not exactly a lot of formatting support in xml docs. 

Can you be more specific about what I'd add.

Reply:
If you want each sentence to appear on a separate like, line in the XML, then you probably need to add `<br />`. I'm just saying I think it's a good idea to check what VS's intellisense shows, and determine whether or not that's easy for customers and partners to read. If not, change it. If it is, then leave it.   I'm not motivated enough to checkout your branch, load the solution, then take a screenshot of intellisense myself.

Reply:
`</br>` didn't really work in my testing, which is why I was asking for suggestions.